nomt/src/store.rs

```diff
@@ -130,3 +150,48 @@ impl Transaction {
         self.batch.put_cf(&cf, b"root", &root[..]);
     }
 }
+
+fn merge_page(_key: &[u8], prev_value: Option<&[u8]>, operands: &MergeOperands) -> Option<Vec<u8>> {
+    let mut val = prev_value.map(|x| x.to_vec());
+    for op in operands.iter() {
+        for op in op.chunks_exact(33) {
+            let val = val.get_or_insert_with(|| vec![0u8; 4096]);
+            let slot_index = op[0] as usize;
+
+            let slot_start = slot_index * 32;
+            let slot_end = slot_start + 32;
+            val[slot_start..slot_end].copy_from_slice(&op[1..]);
+        }
+    }
+
+    // merge operators are not supposed to fail unless there is corruption.
+    val
+}
+
+fn partial_merge_page_ops(
```

Comment:
I would add a comment here explaining that this is merging multiple merge operations into a new one. It's not obvious at first glance that `prev_value` is an older operation and `operands` are possibly multiple new operations

Reply:
yes, good point. I will link to the the RocksDB documentation that makes this clear.